### PR TITLE
@types/xml2js: Replace "Function" with proper callback definition

### PR DESCRIPTION
**Pull request by AHarman**
Please fill in this template.

- [✅] Use a meaningful title for the pull request. Include the name of the package modified.
- [✅] Test the change in your own code. (Compile and run.)
- [✅] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [✅] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [✅] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [✅] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [✅] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://www.npmjs.com/package/xml2js#simple-as-pie-usage
  - https://www.npmjs.com/package/xml2js#options

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## xml2js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/xml2js@latest/))
was missing the following properties:
1. ValidationError
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4923d26199376fc71d57579d7e9d0f9500d2f3e8
</p>

**Review by Grayson-code — APPROVED** [review]

**Review by gabritto — CHANGES_REQUESTED** [review]
The actual change seems good, but some other changes might not be necessary, so I left a comment to check if we really need them.

File: types/xml2js/index.d.ts [hunk 1]
```diff
@@ -46,49 +46,49 @@ export class Builder {
...
-    charsAsChildren?: boolean | undefined;
-    includeWhiteChars?: boolean | undefined;
-    async?: boolean | undefined;
-    strict?: boolean | undefined;
-    attrNameProcessors?: Array<(name: string) => any> | undefined;
-    attrValueProcessors?: Array<(value: string, name: string) => any> | undefined;
-    tagNameProcessors?: Array<(name: string) => any> | undefined;
-    valueProcessors?: Array<(value: string, name: string) => any> | undefined;
-    chunkSize?: number | undefined;
+    attrkey?: string;
```
[the remark this answers is not in the document]
Comment (AHarman) [reply types/xml2js/index.d.ts]: Thanks for raising this, I was mistaken about something and fixed this!

File: types/xml2js/tslint.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
     "extends": "@definitelytyped/dtslint/dt.json",
     "rules": {
-        "ban-types": false
+        "ban-types": false,
+        "no-outside-dependencies": false
```
[the remark this answers is not in the document]
Comment (AHarman) [reply types/xml2js/tslint.json]: I had the same issue documented here: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/49372

```
$ npm run test xml2js

> definitely-typed@0.0.3 test
> dtslint types "xml2js"

dtslint@0.0.111
Error: C:/Users/Alex/Documents/dev/DefinitelyTyped/node_modules/@types/node/assert.d.ts:1:1
ERROR: 1:1  no-outside-dependencies  File C:/Users/Alex/Documents/dev/DefinitelyTyped/node_modules/@types/node/assert.d.ts comes from a `node_modules` but is not declared in this type's `package.json`.  See: https://github.com/Microsoft/dtslint/blob/master/docs/no-outside-dependencies.md
```

However this was only occurring on my local machine and it seems to have passed the check on the CI pipeline.

**Comment by AHarman** [comment]
@gabritto I've made the changes requested

File: types/xml2js/xml2js-tests.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ xml2js.parseStringPromise('', v1Defaults);
 const v2Defaults = xml2js.defaults['0.2'];
 v2Defaults.async = false;
 v2Defaults.chunkSize = 20000;
+v2Defaults.emptyTag = () => {};
```
Comment (72636c) [inline types/xml2js/xml2js-tests.ts:90]:
```suggestion
v2Defaults.emptyTag = () => ({});
```

**Review by 72636c — APPROVED** [review]

**Review by Grayson-code — APPROVED** [review]

**Review by 72636c — APPROVED** [review]

**Comment by 72636c** [comment]
Ready to merge

**Comment by mcdurdin** [comment]
Warning: the version of xml2js on npm/yarn at this time (0.4.23 apparently) does not support `emptyTag` as a function. There is no 0.4.23 tag on the repo. As best as I can tell, the version published to npm is based on commit 8fc5b926846cd4ef9a2dbccd411705e0c110a708.

Comparing to master shows that the only change of significance is supporting emptyTag as a function: https://github.com/Leonidas-from-XIV/node-xml2js/compare/8fc5b926846cd4ef9a2dbccd411705e0c110a708...master

Clearly this should be fixed in xml2js but the repo appears to be unmaintained.

**Pull request merged**